elastic/beats
# heartbeat - move seccomp customization to InitializeModule

## Proposed commit message

Move the overwriting of the arm64 seccomp policy out of the Go init() phase and into a function call that is specific to Heartbeat. This will prevent Heartbeat's policy from affecting all other Beats that run as part of Agentbeat.

Note that other customizations from all Beats that use seccomp.ModifyDefaultPolicy within init() are still being combined in Agentbeat to create a policy that is the aggregation of all seccomp customizations. So the allow-list type policies are likely more permissive than expected.

Fixes #39754

## Checklist




## How to test this PR locally

- Run Hearbeat and observe the seccomp policy that is loaded by enabling debug logging.

```
cd x-pack/heartbeat
go build
./heartbeat run -e -d "seccomp"
```

## Related issues

- Fixes part of #39754

## Logs

Heartbeat on arm64:

```json
{
  "@timestamp": "2024-05-29T19:17:59.332Z",
  "ecs.version": "1.6.0",
  "log.level": "debug",
  "log.logger": "seccomp",
  "log.origin": {
    "file.line": 118,
    "file.name": "seccomp/seccomp.go",
    "function": "github.com/elastic/beats/v7/libbeat/common/seccomp.loadFilter"
  },
  "message": "Loading syscall filter",
  "seccomp_filter": {
    "flag": "tsync",
    "no_new_privs": true,
    "policy": {
      "default_action": "errno",
      "syscalls": [
        {
          "action": "allow",
          "names": [
            "accept",
            "accept4",
            "bind",
            "brk",
            "capget",
            "capset",
            "chdir",
            "clock_gettime",
            "clone",
            "clone3",
            "close",
            "connect",
            "dup",
            "dup3",
            "epoll_create1",
            "epoll_ctl",
            "epoll_pwait",
            "eventfd2",
            "execve",
            "exit",
            "exit_group",
            "faccessat",
            "fadvise64",
            "fallocate",
            "fchdir",
            "fchmod",
            "fchmodat",
            "fchown",
            "fchownat",
            "fcntl",
            "fdatasync",
            "flock",
            "fstat",
            "fstatat",
            "fstatfs",
            "fsync",
            "ftruncate",
            "futex",
            "getcwd",
            "getdents64",
            "getegid",
            "geteuid",
            "getgid",
            "getgroups",
            "getpeername",
            "getpgid",
            "getpid",
            "getppid",
            "getpriority",
            "getrandom",
            "getresgid",
            "getresuid",
            "getrlimit",
            "getrusage",
            "getsockname",
            "getsockopt",
            "gettid",
            "gettimeofday",
            "getuid",
            "inotify_add_watch",
            "inotify_init1",
            "inotify_rm_watch",
            "ioctl",
            "kill",
            "linkat",
            "listen",
            "lseek",
            "madvise",
            "memfd_create",
            "mincore",
            "mkdirat",
            "mlock",
            "mmap",
            "mprotect",
            "munmap",
            "name_to_handle_at",
            "nanosleep",
            "openat",
            "pipe2",
            "ppoll",
            "prctl",
            "pread64",
            "prlimit64",
            "pselect6",
            "pwrite64",
            "read",
            "readlinkat",
            "recvfrom",
            "recvmmsg",
            "recvmsg",
            "renameat",
            "rseq",
            "rt_sigaction",
            "rt_sigprocmask",
            "rt_sigreturn",
            "sched_getaffinity",
            "sched_getattr",
            "sched_getparam",
            "sched_getscheduler",
            "sched_setaffinity",
            "sched_setattr",
            "sched_yield",
            "seccomp",
            "sendfile",
            "sendmmsg",
            "sendmsg",
            "sendto",
            "set_robust_list",
            "set_tid_address",
            "setgid",
            "setgroups",
            "setitimer",
            "setpriority",
            "setsid",
            "setsockopt",
            "setuid",
            "shutdown",
            "sigaltstack",
            "socket",
            "socketpair",
            "splice",
            "statfs",
            "statx",
            "symlinkat",
            "sysinfo",
            "tgkill",
            "tkill",
            "umask",
            "uname",
            "unlinkat",
            "utimensat",
            "wait4",
            "waitid",
            "write",
            "writev"
          ],
          "names_with_args": null
        }
      ]
    }
  },
  "service.name": "heartbeat"
}
```
